components/layout/grid/src/gridItem/index.js

```diff
@@ -1,10 +1,30 @@
 import PropTypes from 'prop-types'
 import cx from 'classnames'
 
-import {BASE_CLASS, CELL_NUMBERS} from '../settings'
+import {BASE_CLASS, CELL_NUMBERS, BREAKPOINTS} from '../settings'
+
+const getColSpanClassNamesTransform = ({colSpan, ...otherProps}) => {
+  const className = []
+  if (CELL_NUMBERS.includes(colSpan) && otherProps.xxs === undefined) {
+    className.push(`${BASE_CLASS}-item--xxs-${colSpan}`)
+  } else if (typeof colSpan === 'object') {
+    Object.values(BREAKPOINTS).forEach(breakpoint => {
+      if (
```

Comment:
maybe you could filter this items out instead of using an if

Reply:
I'll use reduce